transformer_engine/common/comm_gemm_overlap/userbuffers/userbuffers.h

```diff
@@ -19,11 +19,14 @@
 
 #ifdef NVTE_UB_WITH_MPI
 #include <mpi.h>
-typedef MPI_Comm ExtComm;
+#define ExtComm MPI_Comm
 #else
-typedef char *ExtComm;
+#define ExtComm const char *
 #endif
 
+#define ExtAllgatherOp std::function<void(void *, size_t, void *, size_t, ExtComm)>
+#define ExtBarrierOp std::function<void(ExtComm)>
```

Comment:
It would be somewhat nicer to use aliases rather than macros:
```c++
#include <mpi.h>
using ExtComm = MPI_Comm;
#else
using ExtComm = const char *;
#endif

using ExtAllgatherOp = std::function<void(void *, size_t, void *, size_t, ExtComm)>;
using ExtBarrierOp = std::function<void(ExtComm)>;
```
It's basically the same now, but it would be better for the future (e.g. if we defined objects in a namespace instead of in global scope).

Reply:
I agree. This will be especially relevant when we integrate cuBlasMp in the near future.